### ISSUE TITLE
Unable to install any Python version on macOS High Sierra

### ISSUE THREAD

**Issue by thecodejunkie** [pyenv/pyenv#1218]
### Description
- [x] Platform information (e.g. Ubuntu Linux 16.04): macOS High Sierra 10.13.6
- [ ] OS architecture (e.g. amd64): 
- [x] pyenv version: 1.2.7
- [ ] Python version:
- [x] C Compiler information (e.g. gcc 7.3): Apple LLVM version 10.0.0 (clang-1000.11.45.2)
- [x] Please attach verbose build log as gist https://gist.github.com/thecodejunkie/0a12e96f883a512ddb70386ecd0d8b0e

I installed pyenv yesterday (using brew) and I've not been able to install a single python version using it. It keeps failing the build (see gist for verbose pyenv and build logs)

I already have Python 3.7.0 installed using brew and 2.7.0 from the system.

Basically 

```
brew install pyenv
pyenv install 3.7.0 (or any other version)
```

fails. I did check the common build problems and ensures that readline and xz were installed (and updated) as well as a bunch of CFLAGS suggestions I came across while trying to find a solution

Thanks!

**Comment by selfagency** [comment]
i can't build any version on mac os 10.14 mojave either

**Comment by richard-scott** [comment]
Have you ran a `brew update; brew cask upgrade` ?
Google says to change the flag `-std=c99` to `-std=gnu99` when compiling.

**Comment by thecodejunkie** [comment]
@richard-scott brew and brew cask have been updated. Do you mean setting `CFLAGS='-std=gnu99'` ? That has no impact for me

**Comment by mikeybags** [comment]
I'm having a similar issue, though I can't be sure if it's the same since you didn't post the logs. When I run any pyenv install, I get the following:

> VSs-MacBook-Pro-4:.pyenv michael$ pyenv install 3.4.3
> /usr/local/bin/python-build: line 921: !/usr/bin/env: event not found
> /usr/local/bin/python-build: line 923: syntax error near unexpected token `fi'
> /usr/local/bin/python-build: line 921: !/usr/bin/env: event not found
> /usr/local/bin/python-build: line 923: syntax error near unexpected token `fi'
> 
> See all available versions with `pyenv install --list'.
> 
> If the version you need is missing, try upgrading pyenv:
> 
>   brew update && brew upgrade pyenv

I have had almost no luck finding anyone else with this issue, most results point to bash script errors, but I doubt I would have been the only one to see this if that is the case.

I'm setting up a new machine but still have my old one, so I even went as far as to switch to the pyenv version that machine has (1.1.3) to at least get that up and running, but I get the same errors. 

One suggestion was to use 'set +H' to fix the potential bash issue. If I do that, it almost seems like it's going to work, but I get a different error: 

> VSs-MacBook-Pro-4:.pyenv michael$ set +H
> VSs-MacBook-Pro-4:.pyenv michael$ pyenv install 3.4.3
> python-build: use openssl from homebrew
> python-build: use readline from homebrew
> Downloading Python-3.4.3.tar.xz...
> -> https://www.python.org/ftp/python/3.4.3/Python-3.4.3.tar.xz
> Installing Python-3.4.3...
> python-build: use readline from homebrew
> 
> BUILD FAILED (OS X 10.13.6 using python-build 20180424)
> 
> Inspect or clean up the working tree at /var/folders/fl/0xx3pv11755fxym10bvjw4fr0000gq/T/python-build.20181019155816.78918
> Results logged to /var/folders/fl/0xx3pv11755fxym10bvjw4fr0000gq/T/python-build.20181019155816.78918.log
> 
> Last 10 log lines:
>                            ^
>                             void
> In file included from /private/var/folders/fl/0xx3pv11755fxym10bvjw4fr0000gq/T/python-build.20181019155816.78918/Python-3.4.3/Modules/readline.c:32:
> /usr/local/opt/readline/include/readline/readline.h:390:23: warning: this function declaration is not a prototype [-Wstrict-prototypes]
> extern int rl_message ();
>                       ^
>                        void
> 5 warnings generated.
> clang -bundle -undefined dynamic_lookup -L/usr/local/opt/readline/lib -L/usr/local/opt/readline/lib -L/usr/local/opt/openssl/lib -L/Users/michael/Users/michael/.pyenv/Users/michael/.pyenv/versions/3.4.3/lib -L/usr/local/opt/gettext/lib -L/usr/local/opt/readline/lib -L/usr/local/opt/readline/lib -L/usr/local/opt/openssl/lib -L/Users/michael/Users/michael/.pyenv/Users/michael/.pyenv/versions/3.4.3/lib -L/usr/local/opt/gettext/lib -L/usr/local/opt/readline/lib -L/usr/local/opt/readline/lib -L/usr/local/opt/openssl/lib -L/Users/michael/Users/michael/.pyenv/Users/michael/.pyenv/versions/3.4.3/lib -L/usr/local/opt/gettext/lib -I/usr/local/opt/readline/include -I/usr/local/opt/readline/include -I/usr/local/opt/openssl/include -I/Users/michael/Users/michael/.pyenv/Users/michael/.pyenv/versions/3.4.3/include -I/usr/local/opt/gettext/include build/temp.macosx-10.13-x86_64-3.4/private/var/folders/fl/0xx3pv11755fxym10bvjw4fr0000gq/T/python-build.20181019155816.78918/Python-3.4.3/Modules/readline.o -L/usr/lib/termcap -L/usr/local/opt/readline/lib -L/usr/local/opt/openssl/lib -L/Users/michael/Users/michael/.pyenv/Users/michael/.pyenv/versions/3.4.3/lib -L/usr/local/opt/gettext/lib -L/usr/local/lib -lreadline -lncurses -o build/lib.macosx-10.13-x86_64-3.4/readline.so
> ld: warning: directory not found for option '-L/usr/lib/termcap'

Anyone have any suggestions? I'm at a loss at this point.

**Comment by bercikr** [comment]
I'm having the same issue, one thing i noticed is that it seems to be pulling include and lib directories from `/usr/` rather than `/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/` which i believe to be the correct location.

**Comment by bercikr** [comment]
I think this solution works for the issues above (it solved mine) and it should solve @tranthanhbao2207 's issue as well...

https://github.com/pyenv/pyenv/issues/1107

**Comment by thecodejunkie** [comment]
Worked for me!

**Comment by sergeyradov** [comment]
@thecodejunkie  Could you please post solution like steps? 
Is it necessary to close this issue if it is resolved?

**Comment by tranthanhbao2207** [comment]
> I think this solution works for the issues above (it solved mine) and it should solve @tranthanhbao2207 's issue as well...
> 
> #1107
> 
> Basically remove/rename the /usr/include directory...

i re-install the OS and everything works for me ! 
painful tips :) : try to using sudo less

**Comment by tranthanhbao2207** [comment]
Af

> > I think this solution works for the issues above (it solved mine) and it should solve @tranthanhbao2207 's issue as well...
> > #1107
> > Basically remove/rename the /usr/include directory...
> 
> i re-install the OS and everything works for me !
> painful tips :) : try to using sudo less

after several months, i am trying to reinstall another version on python and it has failed with same error 
`from _scproxy import _get_proxy_settings, _get_proxies
ImportError: No module named '_scproxy'
make: *** [install] Error 1`

OS X : 10.13.3

**Comment by ericywl** [comment]
Same problem here, still no solution...

**Comment by eterna2** [comment]
any solutions?

**Comment by liguobao** [comment]
SDKROOT=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk MACOSX_DEPLOYMENT_TARGET=10.15 pyenv install 3.8.0

work for me .

**Comment by ddelange** [comment]
My findings:

SDKROOT is needed. However, ```SDKROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/ pyenv install 3.8.2``` still gave me `ImportError: No module named '_scproxy'` whereas running the command with `sudo` installs successfully. This made me think about the order of paths (`$PATH`/`/etc/paths`).

**Running a bash session without loading my `.bash_profile`/`.bashrc`** allowed me to install without `sudo`. I suspect the culprit to be the line `export PATH="/usr/local/sbin:/usr/local/bin:$PATH"` in my profile.

**Comment by akanksha-sharingan** [comment]
> My findings:
> 
> SDKROOT is needed. However, `SDKROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/ pyenv install 3.8.2` still gave me `ImportError: No module named '_scproxy'` whereas running the command with `sudo` installs successfully. This made me think about the order of paths (`$PATH`/`/etc/paths`).
> 
> **Running a bash session without loading my `.bash_profile`/`.bashrc`** allowed me to install without `sudo`. I suspect the culprit to be the line `export PATH="/usr/local/sbin:/usr/local/bin:$PATH"` in my profile.

The sudo worked for me. Spent almost half day figuring out multiple solutions
